From 8d8efef80d80480b78adbf323e4aa2e2e69a72b1 Mon Sep 17 00:00:00 2001
From: H. Peter Anvin <hpa@zytor.com>
Date: Tue, 16 Sep 2008 09:29:40 -0700
Subject: [PATCH] x86: completely disable NOPL on 32 bits

commit ba0593bf553c450a03dbc5f8c1f0ff58b778a0c8 upstream.

Completely disable NOPL on 32 bits.  It turns out that Microsoft
Virtual PC is so broken it can't even reliably *fail* in the presence
of NOPL.

Signed-off-by: H. Peter Anvin <hpa@zytor.com>

[bwh: Disable use of NOPL directly at the one place the feature flag
is used upstream. This avoids the need to introduce a useless
feature.]
---
 arch/x86/kernel/alternative.c |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/arch/x86/kernel/alternative.c b/arch/x86/kernel/alternative.c
index c36ed19..1a68a4b 100644
--- a/arch/x86/kernel/alternative.c
+++ b/arch/x86/kernel/alternative.c
@@ -157,8 +157,6 @@ static const struct nop {
 } noptypes[] = {
 	{ X86_FEATURE_K8, k8_nops },
 	{ X86_FEATURE_K7, k7_nops },
-	{ X86_FEATURE_P4, p6_nops },
-	{ X86_FEATURE_P3, p6_nops },
 	{ -1, NULL }
 };
 
-- 
1.7.1

